libraries/AP_Mount/AP_Mount_Gremsy.cpp

```diff
@@ -0,0 +1,318 @@
+{
+    // check we have space for the message
+    if (!HAVE_PAYLOAD_SPACE(_chan, GIMBAL_DEVICE_SET_ATTITUDE)) {
+        return;
+    }
+
+    // prepare flags
+    const uint16_t flags = earth_frame ? (GIMBAL_DEVICE_FLAGS_ROLL_LOCK | GIMBAL_DEVICE_FLAGS_PITCH_LOCK | GIMBAL_DEVICE_FLAGS_YAW_LOCK) : 0;
+    const float quat_array[4] = {NAN, NAN, NAN, NAN};
+
+    // send command_long command containing a do_mount_control command
+    mavlink_msg_gimbal_device_set_attitude_send(_chan,
+                                                _sysid,     // target system
+                                                _compid,    // target component
+                                                flags,      // gimbal device flags
+                                                quat_array, // attitude as a quaternion
+                                                roll_rads, pitch_rads, yaw_rads);   // angular velocities
+
+    // store time of send
+    _last_send = AP_HAL::millis();
+}
+
+// send GIMBAL_DEVICE_SET_ATTITUDE to gimbal to control attitude
+// earth_frame should be true if yaw_rad target is in earth frame angle, false if body_frame
+void AP_Mount_Gremsy::send_gimbal_device_set_attitude(float roll_rad, float pitch_rad, float yaw_rad, bool earth_frame)
+{
+    // exit immediately if not initialised
+    if (!_initialised) {
+        return;
+    }
+
+    // check we have space for the message
+    if (!HAVE_PAYLOAD_SPACE(_chan, GIMBAL_DEVICE_SET_ATTITUDE)) {
+        return;
+    }
+
+    // prepare flags
+    const uint16_t flags = earth_frame ? (GIMBAL_DEVICE_FLAGS_ROLL_LOCK | GIMBAL_DEVICE_FLAGS_PITCH_LOCK | GIMBAL_DEVICE_FLAGS_YAW_LOCK) : 0;
+
+    // convert euler angles to quaternion
+    Quaternion q;
+    q.from_euler(roll_rad, pitch_rad, yaw_rad);
+    const float quat_array[4] = {q.q1, q.q2, q.q3, q.q4};
+
+    // send command_long command containing a do_mount_control command
+    mavlink_msg_gimbal_device_set_attitude_send(_chan,
+                                                _sysid,     // target system
+                                                _compid,    // target component
+                                                flags,      // gimbal device flags
+                                                quat_array, // attitude as a quaternion
+                                                NAN, NAN, NAN);   // angular velocities
+
+    // store time of send
+    _last_send = AP_HAL::millis();
+}
+
+// turn motors on/off
+void AP_Mount_Gremsy::enable_motors(bool on)
+{
+    // check we have space for the message
+    if (!HAVE_PAYLOAD_SPACE(_chan, COMMAND_LONG)) {
+        return;
+    }
+
+    // send COMPMAND_LONG with MAV_CMD_USER_1 command with Param7 populated to turn on/off motor
+    mavlink_msg_command_long_send(_chan,
+                                  _sysid,
+                                  _compid,
+                                  MAV_CMD_USER_1,
```

Comment:
wouldn't it make much more sense if this would be done inside the Gremsy gimbal, e.g. that it has an option which allows the user to set the behavior upon retract? If this option would be exposed as MAVLink parameter it could be also done that way, in a much better way. One could also consider adding a respective mode to the MAVLink standard flags. I do see here a generally desired behavior, which strongly suggests against a user_1 command...
If all this fails, it probably would be much more meaning full to have a gremsy mavlink dialect extension. 
IMHO.